### PR TITLE
clean up Makefile

### PR DESCRIPTION
**Pull request by dhellmann**
Try to make the targets simpler and single-purpose. Also update the make target to use the same command as the gosec.sh script.

Addresses #505

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -148,6 +148,10 @@ docker-operator:
 docker-sdk:
 	docker build . -f hack/Dockerfile.operator-sdk
 
+.PHONY: docker-golint
+docker-sdk:
```
Comment (zaneb) [inline Makefile:152]: wrong target name

Comment (dhellmann) [reply Makefile]: paste-o. Fixed.

**Pull request merged**